File: common/src/test/java/com/hartwig/actin/util/json/GsonLocalDateTimeAdapterTest.java
package com.hartwig.actin.util.json;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

import java.time.LocalDateTime;

import com.google.gson.Gson;

import org.junit.Test;

public class GsonLocalDateTimeAdapterTest {

    @Test
    public void shouldSerializeLocalDateTime() {
        LocalDateTime dateTime = LocalDateTime.of(2022, 10, 20, 11, 12, 13);

        Gson serializer = GsonSerializer.create();

        assertEquals(dateTime, serializer.fromJson(serializer.toJson(dateTime), LocalDateTime.class));
        assertNull(serializer.fromJson(serializer.toJson((LocalDateTime) null), LocalDateTime.class));
    }
}

File: common/src/test/kotlin/com/hartwig/actin/util/json/GsonLocalDateTimeAdapterTest.kt
package com.hartwig.actin.util.json

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.time.LocalDateTime

class GsonLocalDateTimeAdapterTest {

    @Test
    fun `Should serialize LocalDateTime object`() {
        val dateTime = LocalDateTime.of(2022, 10, 20, 11, 12, 13)
        val serializer = GsonSerializer.create()
        assertThat(serializer.fromJson(serializer.toJson(dateTime), LocalDateTime::class.java)).isEqualTo(dateTime)
        assertThat(serializer.fromJson(serializer.toJson(null as LocalDateTime?), LocalDateTime::class.java)).isNull()
    }
}